src/components/Card.js

```diff
@@ -1,47 +1,76 @@
-        this._link = data.link;
-        this._cardSelector = cardSelector;
-        this._handleImageClick = handleImageClick;
-        this._element = this._getTemplate();
-        this._likeButton = this._element.querySelector('.card__like-button');
-        this._cardImage = this._element.querySelector('.card__image');
-        this._deleteButton = this._element.querySelector('.card__delete-button');
-        this._setEventListeners();
-    }
+  constructor(data, cardSelector, handleCardClick, handleDeleteClick, api) {
+    this._name = data.name;
+    this._link = data.link;
+    this._id = data._id;
+    this._likes = data.likes || []; 
+    this._isLiked = data.isLiked || false;
+    this._userId = data.userId;
+    this._ownerId = data.owner._id;
+    this._cardSelector = cardSelector;
+    this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._api = api;
+    this._element = this._getTemplate();
+    this._setEventListeners();
+  }
 
-    _getTemplate() {
-        const cardTemplate = document.querySelector(this._cardSelector).content.querySelector(".card");
-        return cardTemplate.cloneNode(true);
-    }
+  _getTemplate() {
+    const cardElement = document
+      .querySelector(this._cardSelector)
+      .content
+      .querySelector('.card')
+      .cloneNode(true);
+    return cardElement;
+  }
 
-    _setEventListeners() {
-        this._likeButton.addEventListener('click', () => this._handleLikeIcon());
-        this._deleteButton.addEventListener('click', () => this._handleDeleteCard());
-        this._cardImage.addEventListener('click', () => this._handleImageClick(this._name, this._link));
-    }
+  _setEventListeners() {
+    this._element.querySelector('.card__delete-button').addEventListener('click', () => {
+      this._handleDeleteClick(this._id, this._element);
+    });
+    this._element.querySelector('.card__image').addEventListener('click', this._handleCardClick);
+    this._element.querySelector('.card__like-button').addEventListener('click', () => this._toggleLike());
+  }
 
-    _handleLikeIcon() {
-        this._likeButton.classList.toggle('card__like-button_active');
+  _toggleLike() {
+    if (this._isLiked) {
+      this._api.removeLike(this._id)
+        .then(updatedCard => {
+          this._isLiked = false;
+          this._likes = updatedCard.likes;
+          this._updateLikesView();
+        })
+        .catch(err => console.error(err));
+    } else {
+      this._api.addLike(this._id)
+        .then(updatedCard => {
+          this._isLiked = true;
+          this._likes = updatedCard.likes;
+          this._updateLikesView();
+        })
+        .catch(err => console.error(err));
     }
+  }
 
-    _handleDeleteCard() {
-        this._element.remove();
-        this._element = null; // Helps with garbage collection
-    }
+  _updateLikesView() {
+    const likeButton = this._element.querySelector('.card__like-button');
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```